target/linux/ramips/dts/mt7628an_oraybox_x1.dts

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "oraybox,x1", "mediatek,mt7628an-soc";
+	model = "OrayBox X1";
+
+	aliases {
+		led-boot = &led_status_white;
+		led-failsafe = &led_status_red;
+		led-running = &led_status_white;
+		led-upgrade = &led_status_red;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_blue: status-blue {
+			label = "blue:status";
+			gpios = <&gpio 37 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_status_red: status-red {
+			label = "red:status";
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_status_white: status-white {
+			label = "white:status";
+			gpios = <&gpio 44 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
```

Comment:
Would be a neat idea to add an entry so that fw_printenv would work.

Reply:
Please add an exntry for thios box to `package/boot/uboot-envtools/files/ramips`. This will allow to read the U-Boot environment.

Reply:
> Please add an entry for this box to `package/boot/uboot-envtools/files/ramips`. This will allow to read the U-Boot environment.

I'm sorry because I got the wrong partition name, the device actually doesn't have a u-boot-env partition, instead it has a kpanic partition, the following is the correct partition information, I will correct it:

[    0.593175] 6 ofpart partitions found on MTD device spi32766.0
[    0.599086] Creating 6 MTD partitions on "spi32766.0":
[    0.604311] 0x000000000000-0x000000030000 : "u-boot"
[    0.611126] 0x000000030000-0x000000040000 : "kpanic"
[    0.618111] 0x000000040000-0x000000050000 : "factory"
[    0.625160] 0x000000050000-0x000000fe0000 : "firmware"
[    0.682096] 2 uimage-fw partitions found on MTD device firmware
[    0.688145] 0x000000050000-0x000000190000 : "kernel"
[    0.694840] 0x000000190000-0x000000fe0000 : "rootfs"
[    0.701727] mtd: device 5 (rootfs) set to be root filesystem
[    0.707636] 1 squashfs-split partitions found on MTD device rootfs
[    0.713924] 0x000000c30000-0x000000fe0000 : "rootfs_data"
[    0.721325] 0x000000fe0000-0x000000ff0000 : "bdinfo"
[    0.803226] 0x000000ff0000-0x000001000000 : "reserve"